src/ucp/core/ucp_context.h

```diff
@@ -39,6 +39,22 @@ UCP_UINT_TYPE(UCP_MAX_LANES)         ucp_lane_map_t;
 #define UCP_MD_LANE_MAP_BITS         64 /* should be UCP_MD_INDEX_BITS * UCP_MAX_LANES */
 UCP_UINT_TYPE(UCP_MD_LANE_MAP_BITS)  ucp_md_lane_map_t;
 
+#if UCP_ENABLE_MT
+#define UCP_CONTEXT_THREAD_CS_ENTER(context_) do {                      \
+        if (context_->config.thread_mode == UCS_THREAD_MODE_MULTI) {    \
+            pthread_mutex_lock(context_->thread_mutex);                 \
+        }                                                               \
+    } while (0)
```

Comment:
no need to have do {..} while(0) because in UCX according to code-stype, if/else statements must be wrapped in { }